packages/@aws-cdk/aws-iotevents/test/detector-model.test.ts

```diff
@@ -137,6 +137,63 @@ test('can set multiple events to State', () => {
   });
 });
 
+test('can set states with transitions', () => {
+  // WHEN
+  const firstState = new iotevents.State({
+    stateName: 'firstState',
+    onEnter: [{
+      eventName: 'test-eventName',
+      condition: iotevents.Expression.fromString('test-eventCondition'),
+    }],
+  });
+  const secondState = new iotevents.State({
+    stateName: 'secondState',
+  });
+
+  firstState.transitionTo({
+    eventName: 'firstToSecond',
+    nextState: secondState,
+    condition: iotevents.Expression.fromString('test-eventCondition-12'),
+  });
+  secondState.transitionTo({
+    eventName: 'secondToFirst',
+    nextState: firstState,
+    condition: iotevents.Expression.fromString('test-eventCondition-21'),
+  });
+
```

Comment:
What if I call `firstState.transitionTo(secondState)` multiple times? Is that allowed? If so, let's add a test for that case. If not, we need to validate that.

Reply:
I confirmed the behavior. And I found that either only one transtion is set if two transiotions provided same target state. When different event name was set, also this behavior did not changed.

So we need to validate it. Thank you for your great insight!!